compiler/infra/ForwardList.hpp

```diff
@@ -0,0 +1,684 @@
+#ifndef FORWARD_LIST_HPP
+#define FORWARD_LIST_HPP
+
+#pragma once
+
+#if 0
```

Comment:
Can we actually extend STL containers? I remember reading somewhere that STL containers do not have virtual destructors and therefore should not be extended.